### PR TITLE
[MINI-4358] Add toggle switch for prod

### PR DESCRIPTION
**Pull request by munir-rakuten**
# Description
Add a toggle switch to switch between prd and stg version.
https://user-images.githubusercontent.com/84305007/143395744-590ee19f-2854-452a-bce9-ce693a183305.mp4

## Links
MINI-4358

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt [hunk 1]
```diff
@@ -26,10 +28,18 @@ import com.rakuten.tech.mobile.testapp.ui.miniapplist.MiniAppListActivity
 import com.rakuten.tech.mobile.testapp.ui.permission.MiniAppDownloadedListActivity
 import com.rakuten.tech.mobile.testapp.ui.settings.MenuBaseActivity.Companion.MENU_SCREEN_NAME
 import com.rakuten.tech.mobile.testapp.ui.userdata.*
+import kotlinx.android.synthetic.main.settings_menu_activity.*
 import kotlinx.coroutines.launch
 import java.net.URL
 import kotlin.properties.Delegates
 
+enum class BuildVariant(name: String){
```
Comment (khairul-alam-licon) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt:36]: Probably, we can move this to `com.rakuten.tech.mobile.testapp` package 🤔

File: testapp/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <string name="lb_app_settings">Settings</string>
     <string name="lb_preview_mode">Preview Mode</string>
     <string name="lb_require_signature_verification">Require Signature Verification</string>
+    <string name="lb_enable_prod_version">PRD</string>
```
Comment (khairul-alam-licon) [inline testapp/src/main/res/values/strings.xml:9]: We can elaborate it to `PROD`, `Prod` or, `Production mode` 🤔

**Review by khairul-alam-licon — COMMENTED** [review]
Nice 👍 
I have left couple of suggestions.

File: testapp/src/main/res/values/strings.xml (hunk 2 above)
Comment (munir-rakuten) [reply testapp/src/main/res/values/strings.xml:9]: @khairul-alam-licon It was written in the ticket description
`The text of the toggle should say 'PRD'`

**Review by khairul-alam-licon — APPROVED** [review]
Looks good 👍

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt [hunk 3]
```diff
@@ -109,7 +114,10 @@ class SettingsMenuActivity : BaseActivity() {
         binding.editParametersUrl.setText(settings.urlParameters)
         binding.switchPreviewMode.isChecked = settings.isPreviewMode
         binding.switchSignatureVerification.isChecked = settings.requireSignatureVerification
-
+        binding.switchProdVersion.isChecked = settings.isProdVersionEnabled
+        if(BuildConfig.BUILD_TYPE == BuildVariant.RELEASE.name){
```
Comment (corycaywood) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt:118]: This toggle switch needs to be available in both the Release and QA builds.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/AppSettings.kt [hunk 4]
```diff
@@ -180,6 +192,18 @@ private class Settings(context: Context) {
...
+                prefs.getBoolean(IS_PROD_VERSION_ENABLED, false)
+            else
+                null
+        set(isRequired) = prefs.edit().putBoolean(IS_PROD_VERSION_ENABLED, isRequired!!).apply()
+
+    var baseUrl: String?
+        get() = prefs.getString(BASE_URL, null)
+        set(baseUrl) = prefs.edit().putString(BASE_URL, baseUrl).apply()
+
     var projectId: String?
```
Comment (corycaywood) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/AppSettings.kt:207]: Don't we also need to switch between the default Prod Project Id/Key and the Staging Id/Key? 

@laubengaier How are you handling this on iOS?

**Review by corycaywood — COMMENTED** [review]
Looking good. I left a couple of comments below.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt (hunk 3 above)
Comment (munir-rakuten) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt:118]: @corycaywood we have `debug`, `release`, `rc` and `staging`.. `release` uses the prod url and keys, so i thought it be better to hide that in release build, so if i show it in release build as well then what will happen if they toggle off/on the switch?

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/AppSettings.kt (hunk 4 above)
Comment (munir-rakuten) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/AppSettings.kt:207]: @corycaywood @laubengaier i thought qa can use id/key by inputing them, if ios has done it automatically i will change too..

**Review by corycaywood — APPROVED** [review]
Looks good 👍

**Pull request merged**